### DISCUSSION TITLE
Custom Car Number

### DISCUSSION THREAD

**Discussion by phenixsims** [kutu/pyirsdk#55] [General]
I don't know if this is an issue or not or something from the iracing telemetry.

I have my car set to be car 64 as my preferred number. I noticed that for a lot of variables in the pyirsdk such as "CarIdxBestLapNum" it's giving me 64 values in an array as the output. Is this the normal behavior? It seems to me that it should only give values for the cars in the race and not extra values for cars that don't exist. I checked the telemetry for spectating and for testing and it's the same no matter what. Lots of extra values that aren't used.

**Comment by kutu** [comment]
car number != car index
you can find your car index from `DriverInfo.Drivers`
then you get `CarIdxBestLapNum[myCarIdx]`

**Reply by phenixsims** [reply]
Do you know why I am getting 64 values for every variable that has the telemetry for all of the current cars?

CarIdxBestLapNum, CarIdxBestLapTime, CarIdxClass, CarIdxClassPosition, CarIdxEstTime, CarIdxF2Time, CarIdxFastRepairsUsed, CarIdxGear, CarIdxLap, CarIdxLapCompleted, CarIdxLapDistPct, CarIdxLastLapTime, CarIdxOnPitRoad, CarIdxP2P_Count, CarIdxP2P_Status, CarIdxPaceFlags, CarIdxPaceLine, CarIdxPaceRow, CarIdxPosition, CarIdxQualTireCompound, CarIdxQualTireCompoundLocked, CarIdxSessionFlags, CarIdxSteer, CarIdxTireCompound, CarIdxTrackSurface, CarIdxTrackSurfaceMaterial

All have arrays like this:
CarIdxClass = [3001, 3001, 3001, 3001, 3001, 3001, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 3001]

Where you can see the first 6 cars belong to other racers and the 64th value is my car.

**Reply by phenixsims** [reply]
DriverInfo = {
            "DriverCarIdx": 63,

This is coming from the telemetry. I don't know where this is coming from or how to fix it.
